libretroshare/src/rsserver/p3webui.cc

```diff
@@ -0,0 +1,206 @@
+/*******************************************************************************
+ * libretroshare/src/rsserver: p3webui.cc                                      *
+ *                                                                             *
+ * libretroshare: retroshare core library                                      *
+ *                                                                             *
+ * Copyright 2019-2019 Cyril Soler                                             *
+ *                                                                             *
+ * This program is free software: you can redistribute it and/or modify        *
+ * it under the terms of the GNU Lesser General Public License as              *
+ * published by the Free Software Foundation, either version 3 of the          *
+ * License, or (at your option) any later version.                             *
+ *                                                                             *
+ * This program is distributed in the hope that it will be useful,             *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                *
+ * GNU Lesser General Public License for more details.                         *
+ *                                                                             *
+ * You should have received a copy of the GNU Lesser General Public License    *
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.       *
+ *                                                                             *
+ *******************************************************************************/
+
+#include "p3webui.h"
+
+#include <thread>
+#include <iostream>
+#include <fstream>
+#include <memory>
+#include <chrono>
+#include <cstdlib>
+#include "util/rsthreads.h"
+#include "util/rsdebug.h"
+#include "retroshare/rswebui.h"
+#include "retroshare/rsjsonapi.h"
+
+#define TEXT_HTML   0
+#define TEXT_CSS    1
+#define TEXT_SVG    2
+#define TEXT_TTF    3
+#define TEXT_WOFF   4
+#define APPLICATION_OCTET_STREAM   5
+
+#define DEBUG_RS_WEBUI 1
+
+RsWebUI *rsWebUI = new p3WebUI;
+
+static constexpr char *mime_types[6] = {
+	"text/html",
+	"text/css",
+	"image/svg+xml",
+	"font/ttf",
+	"font/woff",
+	"application/octet-stream",
+};
+
+#ifdef WINDOWS_SYS
+const std::string RsWebUI::DEFAULT_BASE_DIRECTORY = "data/webui";
+#else
+const std::string RsWebUI::DEFAULT_BASE_DIRECTORY = "/usr/share/retroshare/webui/";
```

Comment:
I think we have an OS aware method to retrieve data directory that work well in all setups, android, windows, mac, linux, appimage etc etc this won't work on all setup for example on android